docs/modules/sql/pages/querying-streams.adoc

```diff
@@ -97,91 +116,190 @@ If the client stops receiving results, but doesn't close the result, Hazelcast w
 
 ```sql
 SELECT *
 FROM TABLE(IMPOSE_ORDER(
-  TABLE trades, <1>
-  DESCRIPTOR(trade_time), <2>
+  TABLE orders, <1>
+  DESCRIPTOR(order_time), <2>
   INTERVAL '0.5' SECONDS) <3>
 );
 ```
 
 <1> The table that contains the events, including the timestamp.
 <2> A pointer to the column that contains the timestamp for the watermark.
-<3> The maximum event lag. Any events that are later than this lag are dropped. For example, an event with a timestamp of `yyyy-mm-dd 23:59:59.5` is added to the window. If another event is processed with a timestamp that's 0.5 seconds or more old, such as ``yyyy-mm-dd 23:59:58.9`, that event is dropped because it is too old.
-
-If an event is delayed by more than the defined maximum event lag, it is dropped.
+<3> The maximum event lag.
 
 NOTE: The above query doesn't currently run in Hazelcast. The `IMPOSE_ORDER()` function must be only used together with `TUMBLE` or `HOP` functions.
 
-For better readability, it's useful to create a view for the watermark like so:
+For better readability, it's useful to create a view:
 
 ```sql
-CREATE VIEW trades_ordered AS
+CREATE VIEW orders_ordered AS
 SELECT *
   FROM TABLE(IMPOSE_ORDER(
-  TABLE trades,
-  DESCRIPTOR(trade_time),
+  TABLE orders,
+  DESCRIPTOR(order_time),
   INTERVAL '0.5' SECONDS)
 );
 ```
 
-We already used this view above. Without the view, you would have to have a nested call to `IMPOSE_ORDER` as the first argument to `TUMBLE`/`HOP` function, which is harder to read.
+Without the view, you would have to have a nested call to `IMPOSE_ORDER` as the first argument to `TUMBLE`/`HOP` function, which is harder to read.
+
+[[joins]]
+== Stream-To-Stream Joins
+
+If you have two or more streams of related data, you can join them together on a related field, process them, and store the result.
+
+The following examples show you how to merge data from an `orders` and a `deliveries` events stream and write this data to a single, aggregated view for querying.  
+
+=== Create Mappings
+
+As for an individual data stream, you must start by creating a mapping for each Kafka topic that you want to use as a data source.
+
+```sql
+CREATE OR REPLACE MAPPING orders (
+  order_id VARCHAR,
+  order_time TIMESTAMP
+  product_id BIGINT)
+TYPE Kafka
+OPTIONS (
+  'valueFormat' = 'json-flat',
+  'bootstrap.servers' = 'kafka:9092'
+);
+```
+```sql
+CREATE OR REPLACE MAPPING deliveries (
+  delivery_id BIGINT,
+  order_id VARCHAR
+  delivery_time TIMESTAMP
+  address TIMESTAMP )
+TYPE Kafka
+OPTIONS (
+  'valueFormat' = 'json-flat',
+  'bootstrap.servers' = 'kafka:9092'
+);
+```
+=== Join the Streams
+
+Next, you need to specify the timebound relationship between the two event streams. This is defined in a SQL `SELECT` statement using a `JOIN`. The conditions of the `JOIN` tell the Jet engine how long to buffer the events from each event stream before processing them; the _postpone_ time. Without this delay, the stream is unlimited, and any queries are rejected. 
```

Comment:
> Without this delay, **the stream is unlimited**, and any queries are rejected.
IMO, it's worth highlighting memory boundness here. The problem is that without these join conditions our internal join processing data structure will not remove any used event it collected before. So, we need a mechanism to clean up memory from overdue events from both JOIN inputs.

Reply:
Let me know if what I've written is not OK.